### ISSUE TITLE
Rule `unicorn/string-content` should allow more "safe" tags

### ISSUE THREAD

**Issue by PopGoesTheWza** [sindresorhus/eslint-plugin-unicorn#602]
this new rule has great potential value, especially with its configurable `patterns` property.

But in some (too many) of my projects I dynamically generate SQL like statements which involves quite a few single quotes.

Would it be possible to add `sql` to the current list of "safe" tags? An even better option would be to make the list of said tags user configurable like `patterns`.

**Comment by sindresorhus** [comment]
Can you share an example on what such statements look like? Are you using tagged template literals?

**Comment by PopGoesTheWza** [comment]
Here is a random pick of where a filter clause is computed. Input is an email address to match with any of two columns.

```typescript
    const parameters = {
      fields: [...fieldNames].join(','),
      filter: `bDepartment=0 AND (${[...value]
        .map(escapeSingleQuote)
        .map((email) => `EMail LIKE '%${email}' OR MailLogin LIKE '%${email}'`)
        .join(' OR ')})`,
    };
```

This one is very basic. Having single quotes in the template is important for two reasons:
- syntax/grammar correctness (though the quoting could be delegated to the tag function)
- visual cue to the dev (who is already managing similar queries with different tools/interfaces like export modules, native client, etc.)

Obviously, I could refactor this, or even turn off the rule.

But since there are "safe" tags (I had totally forgot about this feature until your rule kicked me in the eye) I would rather experiment with tagged templates.

I could "cheat" by taking advantage of the `style.*` safe names... but the possibility of adding custom tags like `dontMessMyQuotes`, `sql`, `sql.*` would be more sensible IMHO.

For reference, link to Micro Focus [REST Api](https://docs.microfocus.com/itom/Asset_Manager:9.71/RestUrlParameters) which is the target in this sample

**Comment by PopGoesTheWza** [comment]
Other common use case of template string requiring the single quote character is when interacting with spreadsheets. Typically using [Google Apps Script](https://developers.google.com/apps-script/) and [Microsoft Graph](https://docs.microsoft.com/en-us/graph/api/resources/excel?view=graph-rest-1.0)
